tests/transformers_/test_transformers_model_export.py

```diff
@@ -0,0 +1,794 @@
+    seq_conf_specify_task = _generate_base_flavor_configuration(
+        small_seq2seq_pipeline, "text-classification"
+    )
+    assert seq_conf_specify_task == expected_seq_pipeline_conf
+    qa_conf_infer_task = _generate_base_flavor_configuration(
+        small_qa_pipeline, _get_or_infer_task_type(small_qa_pipeline)
+    )
+    assert qa_conf_infer_task == expected_qa_pipeline_conf
+    qa_conf_specify_task = _generate_base_flavor_configuration(
+        small_qa_pipeline, "question-answering"
+    )
+    assert qa_conf_specify_task == expected_qa_pipeline_conf
+    with pytest.raises(MlflowException, match="The task provided is invalid. 'magic' is not"):
+        _generate_base_flavor_configuration(small_qa_pipeline, "magic")
+
+
+def test_pipeline_construction_from_base_nlp_model(small_qa_pipeline):
+    generated = _build_pipeline_from_model_input(
+        {"model": small_qa_pipeline.model, "tokenizer": small_qa_pipeline.tokenizer},
+        "question-answering",
+    )
+    assert isinstance(generated, type(small_qa_pipeline))
+    assert isinstance(generated.tokenizer, type(small_qa_pipeline.tokenizer))
+
+
+def test_pipeline_construction_from_base_vision_model(small_vision_model):
+    generated = _build_pipeline_from_model_input(
+        {
+            "model": small_vision_model.model,
+            "image_processor": small_vision_model.image_processor,
+            "tokenizer": small_vision_model.tokenizer,
+        },
+        "image-classification",
+    )
+    assert isinstance(generated, type(small_vision_model))
+    assert isinstance(generated.tokenizer, type(small_vision_model.tokenizer))
+    assert isinstance(generated.image_processor, transformers.MobileNetV2ImageProcessor)
+
+
+def test_pipeline_construction_fails_with_invalid_type(small_vision_model):
+    with pytest.raises(
+        MlflowException, match="The provided model configuration cannot be created as a"
+    ):
+        _build_pipeline_from_model_input(
+            {"model": small_vision_model.feature_extractor}, "image-classification"
+        )
+
+
+def test_saving_with_invalid_dict_as_model(model_path):
+    with pytest.raises(
+        MlflowException, match="Invalid dictionary submitted for 'transformers_model'. The "
+    ):
+        mlflow.transformers.save_model(transformers_model={"invalid": "key"}, path=model_path)
+
+    with pytest.raises(
+        MlflowException, match="The 'transformers_model' dictionary must have an entry"
+    ):
+        mlflow.transformers.save_model(
+            transformers_model={"tokenizer": "some_tokenizer"}, path=model_path
+        )
+
+
+def test_model_card_acquisition_vision_model(small_vision_model):
+    model_provided_card = _fetch_model_card(small_vision_model)
+    assert model_provided_card.data.to_dict()["tags"] == ["vision", "image-classification"]
+    assert model_provided_card.text.startswith(
+        "\n# MobileNet V2\n\nMobileNet V2 model pre-trained on ImageNet-1k"
+    )
```

Comment:
If someone updates the model card of `MobileNet V2`, would this test fail?

Reply:
excellent point. Moved to a length comparison.